common/djangoapps/util/milestones_helpers.py

```diff
@@ -0,0 +1,151 @@
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+from milestones.api import (
+    get_course_milestones,
+    add_milestone,
+    add_course_milestone,
+    remove_course_milestone,
+    get_course_milestones_fulfillment_paths,
+    add_user_milestone,
+    get_user_milestones,
+)
+from django.conf import settings
+
+
+def add_prerequisite_course(course_key, prerequisite_course_key):
+    """
+    It would create a milestone for given course and set it as
+    requirement milestone for given course and set it a fulfilment
+    milestone for pre-requisite course.
+    """
+    # create a milestone
+    milestone = add_milestone({
+        'name': _('Course {} requires {}'.format(unicode(course_key), unicode(prerequisite_course_key))),
+        'namespace': unicode(prerequisite_course_key),
+        'description': _('System defined milestone'),
+    })
+    # add requirement course milestone
+    add_course_milestone(course_key, 'requires', milestone)
+
+    # add fulfillment course milestone
+    add_course_milestone(prerequisite_course_key, 'fulfills', milestone)
+
+
+def remove_prerequisite_course(course_key, milestone):
+    """
+    It would remove pre-requisite course milestone for a course
+    """
+
+    remove_course_milestone(
+        course_key,
+        milestone,
+    )
+
+
+def set_prerequisite_courses(course_key, prerequisite_course_keys):
+    """
+    It would remove any existing requirement milestones for the given `course_key`
+    and create new milestones for each pre requisite course in `prerequisite_course_keys`.
+    To only remove course milestones pass `course_key` and empty list or
+    None as `prerequisite_course_keys` .
+    """
+    #remove any existing requirement milestones with this pre-requisite course as requirement
+    course_milestones = get_course_milestones(course_key=course_key, relationship="requires")
+    if course_milestones:
+        for milestone in course_milestones:
+            remove_prerequisite_course(course_key, milestone)
+
+    # add milestones if pre-requisite course is selected
+    if prerequisite_course_keys:
+        for prerequisite_course_key_string in prerequisite_course_keys:
+            prerequisite_course_key = CourseKey.from_string(prerequisite_course_key_string)
+            add_prerequisite_course(course_key, prerequisite_course_key)
+
+
+def get_pre_requisite_courses_not_completed(user, enrolled_courses):  # pylint: disable=invalid-name
+    """
+    It would make dict of prerequisite courses not completed by user among courses
+    user has enrolled in. It calls the fulfilment api of milestones app and
+    iterates over all fulfillments of user not achieved to make dict of
+    prerequisites yet to be completed.
+    """
+    pre_requisite_courses = {}
+    if settings.FEATURES.get('ENABLE_PREREQUISITE_COURSES'):
+        for course, __ in enrolled_courses:
```

Comment:
`enrolled_courses` is a list returned by `get_course_enrollment_pairs` method in student/views.py. I'm now filtering this list to get only course keys.